### ISSUE TITLE
`softmax_rows` produces NaN for a fully masked row instead of a structured error

### ISSUE THREAD

**Issue by astorise** [astorise/Magnetar#13]
Audited at `222aa7e`.

## Problem

`softmax_rows` (`magnetar-runtime/src/reference_cpu.rs:489`) subtracts the row maximum for numerical stability, which is correct in general but has no guard for a row that is entirely `-inf`:

```rust
let max = slice.iter().copied().fold(f32::NEG_INFINITY, f32::max);
let exponentials = slice
    .iter()
    .map(|value| (value - max).exp())
    .collect::<Vec<_>>();
let sum: f32 = exponentials.iter().sum();
for (col, value) in exponentials.into_iter().enumerate() {
    out[row * cols as usize + col] = value / sum;
}
```

When every element of the row is `f32::NEG_INFINITY`, `max` is `-inf`, so `value - max` is `-inf - (-inf) == NaN`. Every output element for that row becomes `NaN`, and the function returns `Ok`.

## Impact

A fully masked row is the natural representation of "no token is permitted here" — it arises from an all-banned vocabulary mask, from an attention row with an empty visible window, or from logits that a Component's logits processor has fully suppressed. The kernel silently emits `NaN`, which then propagates through downstream matmuls and reaches sampling as a set of non-finite scores. `select_next_token` marks non-finite scores ineligible (`sampling.rs:470`, `eligible: score.is_finite()`) and eventually fails with `SamplingError::NoEligibleToken` — a confusing error attributed to sampling, several kernels away from the row that actually caused it.

This contradicts the stated posture of the Reference CPU Provider, which rejects unsupported cases explicitly elsewhere (see `dtype_conversion` and `layout_conversion` at `reference_cpu.rs:685` and `:704`) rather than producing silent garbage.

## Note on `attention`

The equivalent inline softmax inside `attention` (`reference_cpu.rs:597`) is *not* affected: it reduces over `scores[key_lower..key_upper]`, a range that always contains at least one score computed by the dot-product loop, so `max` is finite there. Only the standalone `softmax_rows` is exposed.

## Suggested fix

Detect the degenerate row and return `ReferenceCpuErrorCode::ShapeUnsupported` (or a new, more precise code):

```rust
if !max.is_finite() {
    return Err(ReferenceCpuError::new(
        ReferenceCpuErrorCode::ExecutionFailed,
        format!("softmax row {row} has no finite entries"),
    ));
}
```

Add a test covering an all-`-inf` row and one covering a row with a single finite entry.
